cmd/cluster-init/generate_job.go

```diff
@@ -0,0 +1,231 @@
+	Tmp            = "tmp"
+)
+
+func generatePeriodic(clusterName string, buildFarmDir string) prowconfig.Periodic {
+	utilityConfig := generateUtilityConfig()
+	utilityConfig.ExtraRefs = append(utilityConfig.ExtraRefs, generateReleaseRef())
+	image := fmt.Sprintf("%s:%s", ApplyConfig, Latest)
+	args := generateArgs(buildFarmDir)
+	args = append(args, fmt.Sprintf("--%s=true", Confirm))
+	container := generateContainer(image, args, generateVolumeMounts())
+	volume := generateSecretVolume(clusterName)
+	podSpec := generatePodSpec(container, []v1.Volume{volume})
+	name := fmt.Sprintf("%s-%s-%s-%s-%s-%s", Periodic, Openshift, Release, Master, clusterName, Apply)
+	jobBase := generateJobBase(podSpec, utilityConfig, name)
+	jobBase.Labels = map[string]string{
+		fmt.Sprintf("%s.%s.%s/%s", Ci, Openshift, Io, Role): Infra,
+	}
+	return prowconfig.Periodic{
+		JobBase:  jobBase,
+		Interval: "12h",
+	}
+}
+
+func generatePostsubmit(clusterName string, buildFarmDir string) prowconfig.Postsubmit {
+	utilityConfig := generateUtilityConfig()
+	image := fmt.Sprintf("%s.%s.%s.%s/%s/%s:%s", Registry, Ci, Openshift, Org, Ci, ApplyConfig, Latest)
+	args := generateArgs(buildFarmDir)
+	args = append(args, fmt.Sprintf("--%s=true", Confirm))
+	container := generateContainer(image, args, generateVolumeMounts())
+	volume := generateSecretVolume(clusterName)
+	podSpec := generatePodSpec(container, []v1.Volume{volume})
+	name := fmt.Sprintf("%s-%s-%s-%s-%s-%s-%s", Branch, Ci, Openshift, Release, Master, clusterName, Apply)
+	jobBase := generateJobBase(podSpec, utilityConfig, name)
+	jobBase.MaxConcurrency = 1
+	jobBase.Labels = map[string]string{
+		fmt.Sprintf("%s.%s.%s/%s", Ci, Openshift, Io, Role): Infra,
+	}
+	brancher := prowconfig.Brancher{
+		Branches: []string{"master"},
+	}
+	return prowconfig.Postsubmit{
+		JobBase:  jobBase,
+		Brancher: brancher,
+	}
+}
+
+func generatePresubmit(clusterName string, buildFarmDir string) prowconfig.Presubmit {
+	utilityConfig := generateUtilityConfig()
+	image := fmt.Sprintf("%s.%s.%s.%s/%s/%s:%s", Registry, Ci, Openshift, Org, Ci, ApplyConfig, Latest)
+	mounts := generateVolumeMounts()
+	mounts = append(mounts, v1.VolumeMount{
+		Name:      Tmp,
+		MountPath: fmt.Sprintf("/%s", Tmp),
+	})
+	container := generateContainer(image, generateArgs(buildFarmDir), mounts)
+	secretVolume := generateSecretVolume(clusterName)
+	emptyVolume := generateEmptyVolume(Tmp)
+	podSpec := generatePodSpec(container, []v1.Volume{secretVolume, emptyVolume})
+	name := fmt.Sprintf("%s-%s-%s-%s-%s-%s-%s", Pull, Ci, Openshift, Release, Master, clusterName, Dry)
+	jobBase := generateJobBase(podSpec, utilityConfig, name)
+	jobBase.Labels = map[string]string{
+		fmt.Sprintf("%s.%s.%s/%s", PjRehearse, Openshift, Io, CanBeRehearsed): "true",
+	}
+	brancher := prowconfig.Brancher{
+		Branches: []string{"master"},
+	}
+	rerun := fmt.Sprintf("/%s %s-%s", Test, clusterName, Dry)
+	trigger := fmt.Sprintf("(?m)^/%s( | .* )%s-%s,?($|\\s.*)", Test, clusterName, Dry)
```

Comment:
You want `prowconfig.DefaultRerunCommandFor` and `prowconfig.DefaultTriggerFor` - check out `prowgen.go#L460`